LockExample/FileService.cs

```diff
@@ -4,49 +4,59 @@
+            }
+            else
+            {
+                DirectoryUtil.EnsureDirectory(dirApp);
+            }
+            var result = new WriteResult(res);
+            if (result.isNotEmptyReadFile())
+            {
+                result.ClearTextFile();
+            }
+            await CopyFileAsync(sourcePath, destinationPath).ConfigureAwait(false);
         }
-        public async Task CopyFileAsync(string source, string destination)
+        public static async Task CopyFileAsync(string source, string destination)
         {
             var fileEntries = Directory.GetFiles(source);
-            var tasks = new List<Task<ResultInfo>>();
+            var tasks = new List<Task>();
             foreach (var item in fileEntries)
             {
                 var task = Task.Run(() => CopyToDestination(source, destination, item));
                 tasks.Add(task);
-                ResultInfo resultItem = await task.ConfigureAwait(false);
-                var writeresult = new WriteResult(_sourcePath);
-                await writeresult.Write(resultItem).ConfigureAwait(false);
             }
-            _ = await Task.WhenAll(tasks).ConfigureAwait(true);
+            await Task.WhenAll(tasks).ConfigureAwait(true);
         }
-        public static Task<ResultInfo> CopyToDestination(string source, string destination, string namefile)
+        public static Task CopyToDestination(string sourcePathFolder, string destinationPathFolder, string namefile)
         {
             var stopWatch = new Stopwatch();
             var currentFileName = Path.GetFileName(namefile);
-            var sourcePath = Path.Combine(source, currentFileName);
-            var destinationPath = Path.Combine(destination, currentFileName);
-            stopWatch.Start();
-            File.Copy(sourcePath, destinationPath, true);
-            stopWatch.Stop();
-            TimeSpan timeCopyFile = stopWatch.Elapsed;
-            //Console.WriteLine($"File {currentFileName} RunTime {Math.Round(timeCopyFile.TotalSeconds * 1000, 2)}");
-            ResultInfo fileInfo = FileResultCopyInfo.GetResultInfo(source, currentFileName, timeCopyFile);
-            return Task.FromResult(fileInfo);
+            var sourcePath = Path.Combine(sourcePathFolder, currentFileName);
+            var destinationPath = Path.Combine(destinationPathFolder, currentFileName);
+            var directoryName = Path.GetDirectoryName(sourcePathFolder) ?? throw new ArgumentException("Error get directory");
+            lock (_lock)
+            {
+                stopWatch.Start();
+                File.Copy(sourcePath, destinationPath, true);
+                stopWatch.Stop();
+                TimeSpan timeCopyFile = stopWatch.Elapsed;
+                ResultInfo fileInfo = FileResultCopyInfo.GetResultInfo(sourcePathFolder, currentFileName, timeCopyFile);
+                var writeresult = new WriteResult(directoryName);
+                writeresult.Write(fileInfo);
```

Comment:
only this part needs to be in lock. Including actual copy to lock again makes execution almost sequential.